sdk/python/kfp/_client.py

```diff
@@ -528,20 +536,24 @@ def schedule_pipeline(self, experiment_id, job_name, pipeline_package_path=None,
     response = self._job_api.create_job(body=schedule_body)
 
 
-  def list_runs(self, page_token='', page_size=10, sort_by='', experiment_id=None):
+  def list_runs(self, page_token='', page_size=10, sort_by='', experiment_id=None, namespace=None):
     """List runs.
     Args:
       page_token: token for starting of the page.
       page_size: size of the page.
       sort_by: one of 'field_name', 'field_name des'. For example, 'name des'.
       experiment_id: experiment id to filter upon
+      namespace: kubernetes namespace to filter upon.
+        For single user deployment, leave it as None;
+        For multi user, input a namespace where the user is authorized.
     Returns:
       A response object including a list of experiments and next page token.
     """
+    namespace = namespace or self.get_user_namespace()
     if experiment_id is not None:
       response = self._run_api.list_runs(page_token=page_token, page_size=page_size, sort_by=sort_by, resource_reference_key_type=kfp_server_api.models.api_resource_type.ApiResourceType.EXPERIMENT, resource_reference_key_id=experiment_id)
     else:
-      response = self._run_api.list_runs(page_token=page_token, page_size=page_size, sort_by=sort_by)
+      response = self._run_api.list_runs(page_token=page_token, page_size=page_size, sort_by=sort_by, resource_reference_key_type=kfp_server_api.models.api_resource_type.ApiResourceType.NAMESPACE, resource_reference_key_id=namespace)
```

Comment:
When both namespace and experiment id is empty, does backend handle this?

Reply:
Backend will error out: https://github.com/kubeflow/pipelines/blob/9f37fd78aafd2919ff34048c2ace48c9afe63511/backend/src/apiserver/server/run_server.go#L75-L99

Reply:
I meant when not in multi user mode.

Reply:
In single user mode, it will return all runs then. There will be no joining with `resource_references` table. We haven't changed any of the single user mode behavior
https://github.com/kubeflow/pipelines/blob/9f37fd78aafd2919ff34048c2ace48c9afe63511/backend/src/apiserver/list/list.go#L235